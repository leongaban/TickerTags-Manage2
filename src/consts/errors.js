export const error401 = 'Error: Request failed with status code 401'
export const invalidLogin = 'Incorrect username or password, please try again.'
export const missingLogin = 'Please fill out both username and password.'
export const notifyDev = 'Something bad happened, notify the dev team!'