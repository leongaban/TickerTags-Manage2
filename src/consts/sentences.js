export const tagline = 'All change is detectable.';