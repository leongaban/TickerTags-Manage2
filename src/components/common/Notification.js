import React from 'react'
import PropTypes from 'prop-types'
import { tagline } from '../../consts/sentences'

const Notification = ({ message, closeMsg }) => {
	return (
		<div className="notification" onClick={ closeMsg }>
			<div className="message">
				<p>{ message }</p>
				<div className="icon-cancel-circled-1"></div>
			</div>
		</div>
	)
}

Notification.propTypes = {
	message: PropTypes.string.isRequired
}

Notification.defaultProps = {
	message: tagline
};

export default Notification