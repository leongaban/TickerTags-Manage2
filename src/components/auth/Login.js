import React from 'react'
import jwt from 'jsonwebtoken'
import { not } from 'ramda'
import { getLoginInputs, loginFormFilled, validEmail } from '../../utils/validation'
import * as session from '../../utils/session'
// import { setCurrentUser } from '../../actions/authActions'
import { error401, invalidLogin, notifyDev, missingLogin } from '../../consts/errors'
import { tagline } from '../../consts/sentences'
import * as api from '../../services/api'
import Notification from '../common/Notification'

class Login extends React.Component {
	constructor(props) {
		super(props)
		// console.log('props', props);

        this.state = {
            username: '',
            password: '',
            error: false,
            errorMsg: ''
        };

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
        this.closeMessage = this.closeMessage.bind(this);
	}

    handleChange(e) {
    	this.setState({ [e.target.name]: e.target.value });
    }

    handleSubmit(e) {
        e.preventDefault();
        const user = this.state.username;
        const pass = this.state.password;

        const gotoDashboard = (res) => {
			const user = res.data.user;
        	const ws_token_cookie = session.getCookie("_ws_token");
        	// console.log('ws_token_cookie', ws_token_cookie);
        	// session.addLocalStorage(ws_token_cookie);
        	// session.setAuthorizationToken(ws_token_cookie);
        	console.log('user', user);
        	// this.props.setUser(user);
        	// dispatch(setCurrentUser(user));

        	// console.log(' jwt', jwt);
        	// console.log('  jwt.decode(ws_token_cookie)', jwt.decode(ws_token_cookie, { complete: true }));
        	this.props.history.push('/dashboard');
        };

        const postLogin = (user, pass) => {
			api.userLogin(user, pass)
				.then((res) => {
					if (res.toString() === error401) {
						return this.setState({
							error: true,
							errorMsg: invalidLogin
						});
					}

					if (res.status && res.status === 200) {
						this.setState({
							error: false,
							errorMsg: ''
						});
						gotoDashboard(res);
					}
					else {
						return this.setState({
							error: true,
							errorMsg: notifyDev
						});
					}
				});
		};

        loginFormFilled(user, pass)
        	? postLogin(user, pass)
        	: this.setState({
        		error: true,
        		errorMsg: missingLogin
        	});
    }

    closeMessage() {
    	this.setState({ error: false, errorMsg: '' });
    }

	render() {
		const usernameError = this.state.username.error;
		const error = this.state.error;
		const errorMsg = this.state.errorMsg;

		return (
			<div className="app-bg">
				{ error &&
					<Notification message={ errorMsg } closeMsg={ this.closeMessage }/>
				}
                <section id="auth-section">
                	<header>
	                	<img src="static/imgs/logo.png"/>
	                	<h1>{ tagline }</h1>
                	</header>
                	<div className="login-form">
	                	<form onSubmit={ this.handleSubmit } className="form-login">
	                        <ul>
	                            <li className="rel">
	                            	<div className="icon-user-circle-o"></div>
	                                <input type="text"
	                                       id="input-auth-username"
	                                       name="username"
	                                       placeholder="username"
	                                       onChange={ this.handleChange } />
	                            </li>
	                            <li className="rel">
	                            	<div className="icon-lock"></div>
	                                <input type="password"
	                                       id="input-auth-password"
	                                       name="password"
	                                       placeholder="password"
	                                       onChange={ this.handleChange } />
	                            </li>
	                            <li>
	                                <button className="btn-green" type="submit">Login</button>
	                            </li>
	                            <li>
	                                <a href="/forgot" className="small-auth-link">
	                                    <p>Forgot Password</p>
	                                </a>
	                                <footer> &copy; { new Date().getFullYear() } TickerTags</footer>
	                            </li>
	                        </ul>
	                    </form>
					</div>
                </section>
			</div>
		)
	}
}

export default Login