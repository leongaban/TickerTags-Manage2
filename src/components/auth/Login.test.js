import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { missingLogin } from '../../consts/errors'
import Login from './Login'
import Notification from '../common/Notification'

const user = {
	id: 1,
	role: 'Admin',
	username: 'leongaban'
};

const loginComponent = shallow(
	<Login/>
);

const fakeEvent = { preventDefault: () => '' };

describe('<Login /> component', () => {
	it('should render', () => {
		const tree = toJson(loginComponent);
		expect(tree).toMatchSnapshot();
	});

	it('should contain the words "Forgot Password"', () => {
		expect(loginComponent.contains('Forgot Password')).toBe(true);
	});

	it('should render the Notification component if state.error is true', () => {
		loginComponent.setState({ error: true });
		expect(loginComponent.find(Notification).length).toBe(1);
	});
});

describe('User Login', () => {
	it('should fail if no credentials are provided', () => {
		expect(loginComponent.find('.form-login').length).toBe(1);
		loginComponent.find('.form-login').simulate('submit', fakeEvent);
		expect(loginComponent.find(Notification).length).toBe(1);
		const notificationComponent = shallow(<Notification message={ missingLogin }/>);
		expect(notificationComponent.text()).toEqual('Please fill out both username and password.');
	});

	it('input fields should be filled correctly', () => {
		const credentials = { username: 'leongaban', password: 'testpass' };
		expect(loginComponent.find('#input-auth-username').length).toBe(1);

		const usernameInput = loginComponent.find('#input-auth-username');
		usernameInput.value = credentials.username;
		expect(usernameInput.value).toBe('leongaban');
		
		const passwordInput = loginComponent.find('#input-auth-password');
		passwordInput.value = credentials.password;
		expect(passwordInput.value).toBe('testpass');
	});
});