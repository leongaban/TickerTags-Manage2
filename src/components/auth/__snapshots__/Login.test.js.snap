// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> component should render 1`] = `
<div
  className="app-bg"
>
  <section
    id="auth-section"
  >
    <header>
      <img
        src="static/imgs/logo.png"
      />
      <h1>
        All change is detectable.
      </h1>
    </header>
    <div
      className="login-form"
    >
      <form
        className="form-login"
        onSubmit={[Function]}
      >
        <ul>
          <li
            className="rel"
          >
            <div
              className="icon-user-circle-o"
            />
            <input
              id="input-auth-username"
              name="username"
              onChange={[Function]}
              placeholder="username"
              type="text"
            />
          </li>
          <li
            className="rel"
          >
            <div
              className="icon-lock"
            />
            <input
              id="input-auth-password"
              name="password"
              onChange={[Function]}
              placeholder="password"
              type="password"
            />
          </li>
          <li>
            <button
              className="btn-green"
              type="submit"
            >
              Login
            </button>
          </li>
          <li>
            <a
              className="small-auth-link"
              href="/forgot"
            >
              <p>
                Forgot Password
              </p>
            </a>
            <footer>
               Â© 
              2017
               TickerTags
            </footer>
          </li>
        </ul>
      </form>
    </div>
  </section>
</div>
`;
